sw/device/silicon_creator/rom/rom_isrs.S

```diff
@@ -0,0 +1,89 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey_memory.h"
+#include "sw/device/lib/base/macros.h"
+#include "flash_ctrl_regs.h"
+
+.equ LOAD_ACCESS_FAULT, 5
+
+// If we get an ECC error reading from internal flash, one of these bits
+// will be set in the flash controller's FAULT_STATUS register.
+.equ PHY_ERRORS, (1 << FLASH_CTRL_FAULT_STATUS_PHY_STORAGE_ERR_BIT) | \
+                 (1 << FLASH_CTRL_FAULT_STATUS_PHY_RELBL_ERR_BIT)
+
+// NOTE: The "ax" flag below is necessary to ensure that this section
+// is allocated executable space in ROM by the linker.
+.section .rom_isrs, "ax"
+
+// -----------------------------------------------------------------------------
+
+  /**
+   * Exception handler.
+   */
+  .balign 4
+  .global rom_exception_handler_entry
+  .type rom_exception_handler_entry, @function
+rom_exception_handler_entry:
+  // Save all registers to the stack.
+  addi sp, sp, -OT_CONTEXT_SIZE
+  sw   t0,  2 * OT_WORD_SIZE(sp)
```

Comment:
Do we have the shadow stack enabled in the ROM? If so, @luismarques can provide guidance on what would be the best way to use it in this routine.